### PR TITLE
feat: file:// URLs support

### PR DESCRIPTION
**Pull request by diivi**
Yet to do (and write) some tests.

### CODE REVIEW

File: borgmatic/borg/export_tar.py [hunk 1]
```diff
@@ -47,7 +47,11 @@ def export_tar_archive(
         + (('--tar-filter', tar_filter) if tar_filter else ())
         + (('--strip-components', str(strip_components)) if strip_components else ())
         + flags.make_repository_archive_flags(
-            repository if ':' in repository else os.path.abspath(repository),
+            os.path.abspath(repository)
+            if ':' not in repository
+            else os.path.abspath(repository[7:])
```
Comment (witten) [inline borgmatic/borg/export_tar.py:52]: Here's a way to do this last part that avoids a magic number (or at least uses a slightly less magic one):

```python
os.path.abspath(repository.partition('file://')[-1])
```

An alternative:

```python
os.path.abspath(repository[len('file://'):])
```

File: borgmatic/borg/export_tar.py [hunk 2]
```diff
@@ -47,7 +47,11 @@ def export_tar_archive(
...
+            if repository.startswith('file://')
+            else repository,
```
[the remark this answers is not in the document]
Comment (diivi) [reply borgmatic/borg/export_tar.py:54]: Yes, I can.
I had the same doubt but my reason for doing this was that even before my changes there was a conditional here (on top of the same one in normalize.py).
Was that also redundant?

File: borgmatic/borg/export_tar.py (hunk 1 above)
Comment (diivi) [reply borgmatic/borg/export_tar.py:52]: yeah, I used the second suggestion before I used the magic number, but got some interesting conflicts between black and flake8 😅. Will share on IRC and try the first suggestion.

File: borgmatic/borg/export_tar.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (diivi) [reply borgmatic/borg/export_tar.py:54]: yeah, looks like it was redundant at the first place, I'll use `repository` directly and check.

[the remark this answers is not in the document]
Comment (witten) [reply borgmatic/borg/export_tar.py:54]: I think you are correct that it's redundant. My guess is that `normalize.py` wasn't always doing `abspath()` and so, previously, these action functions needed to do it themselves.

**Comment by witten** [comment]
Thanks for making the revisions! This looks good to me.

**Comment by witten** [comment]
Heads up that I removed an unused import in this that flake8 was complaining about (`os` in `export_tar.py`), which subsequently broke tests that were relying on it. These tests were setting up expectations on `os.path.abspath()`, which wasn't used any longer in `export_tar.py` after your changes. So I just removed those expectations and now tests pass again!

**Pull request merged**